File: 프로그래머스/lv3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    var answer = 0;
    let graph = Array.from(Array(n + 1), () => Array());
    let check = Array(n + 1).fill(true);
    for(let i=0;i<n;i++){
        for(let j=0;j<n;j++){
            if(computers[i][j]===1) graph[i+1].push(j+1)
        }
    }
    function DFS(v) {
        for (let x of graph[v]) {
          if (check[x]) {
            check[x] = false;
            DFS(x);
          }
        }
      }
      for (let i = 1; i <= n; i++) {
        if (check[i]) {
          answer++;
          check[i] = false;
          DFS(i);
        }
      }
    return answer;
}

File: 프로그래머스/lv3/43162. 네트워크/네트워크.py
import copy
def solution(n, computers):
    graph = [[] for _ in range(n+1)]
    res = []
    for k in range(n):
        for u in range(n):
            if computers[k][u] == 1 and k!=u:
                graph[k].append(u)
    print(graph)
    
    def dfs(graph,v,visited):
        visited[v] = True
        for k in graph[v]:
            if not visited[k]:
                dfs(graph,k,visited)
    
    for k in range(n):
        graph2 = copy.deepcopy(graph)
        visited = [False]*n
        dfs(graph2,k,visited)
        if visited not in res:
            res.append(visited)
    
    return len(res)